k9mail-library/src/main/java/com/fsck/k9/mail/ssl/DefaultTrustedSocketFactory.java

```diff
@@ -203,6 +223,11 @@ public Socket createSocket(Socket socket, String host, int port, String clientCe
 
         setSniHost(socketFactory, sslSocket, host);
 
+        if (!proxySettings.enabled && socket == null) {
```

Comment:
Yes I think it does. Because the proxied case has to return a connected socket, all the methods do. I think this is okay but the method name should indicate that this is the case.